### ISSUE TITLE
Warn before opening large files

### ISSUE THREAD

**Issue by darealshinji** [notepadqq/notepadqq#130]
I have a feature request:
When notepadqq opens a file which exceeds a limit specified in the settings, instead of opening the file it should issue a warning and ask the user for permission to load the file. And only if the user confirms it the file is loaded.
This would be very useful if you (accidentally) open a text file of the size of several MB on a system with not so much RAM left.

**Comment by danieleds** [comment]
I agree!
Ideally, a good instantaneous limit should be calculated in function of the current available ram.

**Comment by gnrdeepak** [comment]
Hi, is there a way to disable this or set the limit when notepadqq can prompt the warning?

**Comment by Teklad** [comment]
@gnrdeepak Could you elaborate? Are you wanting a setting that sets the maximum file size or something along those lines?

**Comment by mdealer** [comment]
This feature should be opt-in rather than standard. I've got 32GB of RAM and it warns me about small files (3MB).

**Comment by klitoskyriacou** [comment]
I agree with mdealer. Even with 100s of MB it doesn't take *that* long to open the files, and warning about files larger than 1 MB is annoying. As I found out by looking through the source code, it *is* possible to change this. Add the following line in the General section of ~/.config/Notepadqq/Notepadqq.ini to warn only if files are more than 30 MiB:
`WarnIfFileLargerThan=30
`
